packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -0,0 +1,1005 @@
+											>
+												<VStack
+													spacing="3"
+													className="editor-collab-sidebar__replyComment"
+												>
+													<TextareaControl
+														className="editor-collab-sidebar__replyComment__textarea"
+														value={
+															commentReply ?? ''
+														}
+														onChange={ (
+															value
+														) => {
+															setCommentReply(
+																value
+															);
+														} }
+													/>
+													<VStack
+														alignment="left"
+														spacing="3"
+														justify="flex-start"
+														onRequestClose={ () =>
+															setHasCommentReply(
+																false
+															)
+														}
+													>
+														<HStack
+															alignment="left"
+															spacing="3"
+															justify="flex-start"
+															className="editor-collab-sidebar__replybtn"
+														>
+															<Button
+																variant="primary"
+																onClick={ () => {
+																	confirmReplyComment(
+																		reply.id
+																	);
+																	setHasCommentReply(
+																		false
+																	);
+																} }
+																className="is-compact"
+															>
+																{ __(
+																	'Reply'
+																) }
+															</Button>
+															<Button
+																onClick={ () => {
+																	setHasCommentReply(
+																		false
+																	);
+																	setShowConfirmation(
+																		false
+																	);
+																} }
+																className="is-compact"
+															>
+																{ __(
+																	'Cancel'
+																) }
+															</Button>
+														</HStack>
+													</VStack>
+												</VStack>
+											</HStack>
+										) }
+
+									{ commentConfirmation &&
+										reply.id === showConfirmationTabId && (
+											<Text>
+												{ __(
+													'Thread marked as resolved.'
+												) }
+											</Text>
+										) }
+									{ commentEditedMessage &&
+										reply.id === showConfirmationTabId && (
+											<Text>
+												{ __(
+													'Thread edited successfully.'
+												) }
+											</Text>
+										) }
+									{ commentDeleteMessage &&
+										reply.id === showConfirmationTabId && (
+											<Text>
+												{ __(
+													'Thread deleted successfully.'
+												) }
+											</Text>
+										) }
+									{ replyMessage &&
+										reply.id === showConfirmationTabId && (
+											<Text>
+												{ __(
+													'Reply added successfully.'
+												) }
+											</Text>
+										) }
+
+									{ showConfirmation &&
+										reply.id === showConfirmationTabId && (
+											<VStack
+												title={ __( 'Confirm' ) }
+												onRequestClose={ () =>
```

Comment:
I don't think `VStack` accepts a `onRequestClose` prop?

Reply:
We have refactored the code and modularized it into components.